proto/plugin.proto

```diff
@@ -677,9 +720,48 @@ message Deploy {
 
     // template data for the deployment
     bytes template_data = 2;
+
+    // Izaak: testing
+    DeclaredResources declared_resources = 4;
   }
 }
 
+// A platform resource that an operation (release/deployment) has created, depends on, or manages.
+message DeclaredResource {
+  // platform-specific name of the resource type. i.e. instance, pod, auto-scaling group, etc
+  string type = 1;
+
+  // The platform on which the resource should exist, i.e. docker, gcp, k8s, etc.
+  string platform = 2;
+
+  // Internal ResourceManager representation of the resource.
+  google.protobuf.Any state = 3;
```

Comment:
Should this have a more resource manager specific name so it's clear that's what should be set here? I can see people accidentally setting this to something non-resource manager specific based on the name and the fact that it's an `Any` proto 🤔

Reply:
I actually changed this from `resource_state` to just `state` based on @mitchellh 's feedback. The argument for keeping this generic is that the server doesn't care what's in this field - a plugin can store whatever state it wants in here. A future SDK may support additional frameworks or changes that may override that state field.

Plugin devs also won't need to fill this in manually if they're using resource manager, so I don't think it'll be too confusing in practice.